### PR TITLE
*: merge hot region schedulers

### PR DESCRIPTION
**Pull request by nolouch**
merge hot write region and hot read region scheduler

### CODE REVIEW

**Comment by ngaut** [comment]
Could you share performance details?

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/server/core"
+	"github.com/pingcap/pd/server/schedule"
+)
+
+func init() {
+	schedule.RegisterScheduler("hotRegion", func(opt schedule.Options, args []string) (schedule.Scheduler, error) {
```
Comment (siddontang) [inline server/schedulers/hot_region.go]: I think we should change all name to kebab-case later. 

/cc @disksing

**Comment by siddontang** [comment]
This PR just removes duplicated code for read and write hot schedulers. No performance gain.

**Comment by siddontang** [comment]
PTAL @zhangjinpeng1987

File: server/schedulers/hot_region.go (hunk 1 above)
Comment (disksing) [reply server/schedulers/hot_region.go]: @Connor1996 will take care of it.

**Comment by disksing** [comment]
LGTM.

**Pull request merged**